configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -690,6 +699,13 @@ private static void logConfigChangeActions(ConfigChangeActions actions, DeployLo
         }
     }
 
+    private String getLogServerHostname(ApplicationId applicationId) {
+        Application application = getApplication(applicationId);
+        VespaModel model = (VespaModel) application.getModel();
+        String logServerHostname = model.getAdmin().getLogserver().getHostName();
```

Comment:
I think you need to actually check if there is a container on the same host and find the port used by it (use Application.getHosts() and look at HostInfo and ServiceInfo to see if one exists) from that. You can probably throw IllegalArgumentException if no container exists (it will end up as a 404)